applications/RomApplication/python_scripts/hrom_training_utility.py

```diff
@@ -206,19 +195,67 @@ def __AppendHRomWeightsToRomParameters(self):
             # Add the missing parents to the elements dict with a null weight
             for parent_id in missing_condition_parents:
                 hrom_weights["Elements"][parent_id] = 0.0
+            w, z = self.__AddSelectedElementsWithZeroWeights(w,z, missing_condition_parents)
+
+
 
-        # Append weights to RomParameters.json
-        # We first parse the current RomParameters.json to then append and edit the data
-        with open('RomParameters.json','r') as f:
-            updated_rom_parameters = json.load(f)
-            #FIXME: I don't really like to automatically change things without the user realizing...
-            #FIXME: However, this leaves the settings ready for the HROM postprocess... something that is cool
-            #FIXME: Decide about this
-            # updated_rom_parameters["train_hrom"] = False
-            # updated_rom_parameters["run_hrom"] = True
-            updated_rom_parameters["elements_and_weights"] = hrom_weights #TODO: Rename elements_and_weights to hrom_weights
+        if self.hrom_output_format=="numpy":
+            np.save('WeightsMatrix.npy',w)
+            np.save('ElementsVector.npy',z)
 
-        with open('RomParameters.json','w') as f:
-            json.dump(updated_rom_parameters, f, indent = 4)
+        elif self.hrom_output_format=="json":
+            with open('RomParameters.json','r') as f:
+                updated_rom_parameters = json.load(f)
+                updated_rom_parameters["elements_and_weights"] = hrom_weights #TODO: Rename elements_and_weights to hrom_weights
```

Comment:
In the json format case, you update the rom parameters locally but you're not saving them again as a json anymore. Is that alright?

Reply:
You are right. Its been added